src/solvers/dg/2d/dg.jl

```diff
@@ -1,8 +1,9 @@
 # Main DG data structure that contains all relevant data for the DG solver
 mutable struct Dg2D{Eqn<:AbstractEquation, NVARS, POLYDEG,
                   SurfaceFlux, VolumeFlux, InitialConditions, SourceTerms,
-                  MortarType, VolumeIntegralType,
+                  MortarType, VolumeIntegralType, ShockIndicatorVariable,
                   VectorNnodes, MatrixNnodes, MatrixNnodes2,
+                  InverseVandermondeLegendre, MortarMatrix,
```

Comment:
Why add `MortarMatrix` if it seems to be the same as `MatrixNnodes`? Or is it not?

Reply:
I needed the additional parameter for comparing different matrix types there. If we keep it like this, it is easier for users to switch to other matrix types, which might be more efficient for higher polynomial degrees or something like that. Or shall I get rid of that stuff and hard-code the types that are optimized for a specific test problem we have in our zoo?

Reply:
By now I almost don't care anymore. According to your comment above, some of these types we'll get rid of when we move to TaaL (Trixi-as-a-library), so we can revisit this in the future.

Reply:
Okay, sounds good to me. We'll keep things like that in our mind and decide what to do when the next big rounds of refactoring are finished.